src/docs/binding-generics.md

```diff
@@ -437,3 +437,191 @@ We could also consider solving this on the runtime level.
 ### Generic classes
 
 TBD
+
+## Protocol constraints
+
+This section describes how to project protocol constraints on generic type parameters from Swift to C\#. The goal here is not to describe full projections of protocols; for that, please refer to the [binding protocols doc](binding-protcols.md).
+
+A Swift protocol can be mapped to C\# as an interface. Then Swift generic parameters with protocol constraint can be mapped into C\# as generic parameters with interface constraint. This should work for both functions and types.
+
+### Protocol constraints on functions
+
+Consider the following simple example in Swift:
+
+```swift
+public protocol Printable {
+ func printMe()
+}
+
+public func printPrintable<T: Printable>(data: T) {
+ data.printMe()
+}
+
+// Equivalent to: public func printPrintable<T>(data: T) where T : Printable
+```
+
+This can be projected into C\# as follows:
+
+```csharp
+public interface ISwiftPrintableProtocol
+{
+    public void PrintMe();
+    public abstract static NativeHandle GetISwiftPrintableProtocolWitnessTable();
+
+}
+
+public static void PrintPrintable<T>(T data) where T : ISwiftObject, ISwiftPrintableProtocol
+{
+    var metadata = TypeMetadata.GetTypeMetadataOrThrow<T>();
+    var protocolWitnessTable = T.GetISwiftPrintableProtocolWitnessTable();
+    PInvokePrintPrintable(data.GetPayload(), metadata, protocolWitnessTable);
+}
+
+[DllImport(Path, EntryPoint = "...")]
+[UnmanagedCallConv(CallConvs = [typeof(CallConvSwift)])]
+private static extern void PInvokePrintPrintable(NativeHandle data, TypeMetadata metadata, NativeHandle protocolWitnessTable);
+```
+
+The only ABI difference is that some Swift function will expect to be passed Protocol Witness Table for the generic types at the end of the argument list.
```

Comment:
See above - it's slightly more complicated. IIRC, if you have this:
```swift
public func foo<T: P1, U:P2>(a: T, b:U) { }
```
Then then the arguments in order are:
1. pointer to a
2. pointer to b
3. type metadata for T
4. protocol witness table for P1
5. type metadata for U
6. protocol witness table for P2

Reply:
Yes, this is correct - I just double checked it.